plugins/pagerduty/src/components/PagerDutyCard/index.tsx

```diff
@@ -67,6 +72,17 @@ export const PagerDutyCard = () => {
     loading,
     error,
   } = useAsync(async () => {
+    if (serviceId) {
```

Comment:
This is my crude attempt at switching the behaviour, but Im not happy with it.

Reply:
Yeah alright I'd say the existing API doesn't really make this easy for you 😅 

Tbh I think the best way forward here would be to refactor the API so that it's more oriented around fetching things tied to entities. Thinking `getServiceByEntity(entity: Entity)` or that kind of thing